### DISCUSSION TITLE
Guest scenario

### DISCUSSION THREAD

**Discussion by ISulimanov** [zitadel/zitadel#5304] [Q&A] [answered]
Is it possible to implement some kind of guest scenario via Zitadel? 

Lets say system generates UUID for user before he proceed registration.
User has account with specific role until registration not complete (email not entered)

It's like "continue without email" 

May be I can achieve it by creating temp users and custom actions but not sure, any advice?

**Comment by fforootd** [comment, marked as answer]
You mean something like a "anonymous session" like[ firebase does](https://firebase.google.com/docs/auth/web/anonymous-auth)?

The answer would be not at the moment 😁 but any inputs would be welcome in our tracking issue here https://github.com/zitadel/zitadel/issues/5016
